feat: add web route for sending WOL package to given mac addr

tadic-luka opened the pull request:
# Summary:
adds a new route `/device/send-wol/:mac-addr` to send the magic WOL package to the specified mac-addr.
Method is POST and is protected.
Internally, the url handler uses already existing function from wol.go called rpcSendWOLMagicPacket. 


Useful for custom wake up scripts: example is sending HTTP request through iOS shortcut to wake up a server.

I could also add a feature to specify the name of the device set up in the WOL UI if needed.

# Test plan:
calling the API with curl with valid mac addr
```
$ curl -X POST   http://<jetkvm-ip>/device/send-wol/xx:xx:xx:xx:xx:xx
WOL sent to xx:xx:xx:xx:xx:xx
```

and observing the magic packet on my laptop/PC:
```
$ ncat -u -l 9 -k | xxd
00000000: ffff ffff ffff d050 9978 a620 d050 9978  .......P.x. .P.x
00000010: a620 d050 9978 a620 d050 9978 a620 d050  . .P.x. .P.x. .P
00000020: 9978 a620 d050 9978 a620 d050 9978 a620  .x. .P.x. .P.x.
00000030: d050 9978 a620 d050 9978 a620 d050 9978  .P.x. .P.x. .P.x
00000040: a620 d050 9978 a620 d050 9978 a620 d050  . .P.x. .P.x. .P
00000050: 9978 a620 d050 9978 a620 d050 9978 a620  .x. .P.x. .P.x.
```

Calling the API with curl with invalid mac addr:
```
curl -X POST   http://<jetkvm-ip>/device/send-wol/22 -v
...
> POST /device/send-wol/22 HTTP/1.1
...
* Request completely sent off
< HTTP/1.1 400 Bad Request
< Content-Type: text/plain; charset=utf-8
...
Invalid mac address provided
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/jetkvm/kvm?pullRequest=945) <br/>All committers have signed the CLA.

IDisposable commented on web.go:
```diff
@@ -807,3 +809,12 @@ func handleSetup(c *gin.Context) {
 }
+
+func handleSendWOLMagicPacket(c *gin.Context) {
+	macAddr := c.Param("mac-addr")
```
Would prefer there be some validation on this raw path parameter (is this a reasonable MAC address) before calling the WOL.

tadic-luka replied:
@IDisposable  done! I also update the test plan and resolved the issue with CLA.

IDisposable reviewed (changes requested):
Other than to minor validation, this looks great! Thanks for the submission. Make sure you get the CLA agreement sorted, please.

tadic-luka committed:
feat: add web route for sending WOL package to given mac addr

```

adds a new route /device/send-wol/:mac-addr to send the magic WOL package
to the specified mac-addr.
Method is POST and is protected.

Useful for custom wake up scripts: example is sending HTTP request through iOS shortcut

Test plan:
calling the API with curl
```
$ curl -X POST   http://<jetkvm-ip>/device/send-wol/xx:xx:xx:xx:xx:xx
WOL sent to xx:xx:xx:xx:xx:xx
```

and observing the magic packet on my laptop/PC:
```
$ ncat -u -l 9 -k | xxd
00000000: ffff ffff ffff d050 9978 a620 d050 9978  .......P.x. .P.x
00000010: a620 d050 9978 a620 d050 9978 a620 d050  . .P.x. .P.x. .P
00000020: 9978 a620 d050 9978 a620 d050 9978 a620  .x. .P.x. .P.x.
00000030: d050 9978 a620 d050 9978 a620 d050 9978  .P.x. .P.x. .P.x
00000040: a620 d050 9978 a620 d050 9978 a620 d050  . .P.x. .P.x. .P
00000050: 9978 a620 d050 9978 a620 d050 9978 a620  .x. .P.x. .P.x.
```

calling the api with invalid mac addr returns HTTP 400 error
```
$ curl -X POST -v http://<jetkvm-ip>/device/send-wol/abcd
...
* Request completely sent off
< HTTP/1.1 400 Bad Request
...
...
Invalid mac address provided

tadic-luka commented:
Updated commit:
- added mac address validation
- updated author email in the commit to fix the CLA issue

tadic-luka commented:
@IDisposable apologies if I am making too much notfications, still getting used to GitHub workflow.

IDisposable commented:
> @IDisposable apologies if I am making too much notfications, 

There's no such thing as too much information ;)

> still getting used to GitHub workflow.

You're rocking it!

IDisposable committed:
Resolve golint complaint

IDisposable commented:
I resolved the trivial golint issue (which directly changed your branch), make sure you FETCH/PULL to get current before doing more.

Also, for future ease on your side, I recommend not doing edits directly against the `dev` branch, you should create a branch for the feature/fix you're working.

IDisposable commented on web.go:
```diff
@@ -807,3 +810,23 @@ func handleSetup(c *gin.Context) {
+	inputMacAddr := c.Param("mac-addr")
+	macAddr, err := net.ParseMAC(inputMacAddr)
+	if err != nil {
+		logger.Warn().Err(err).Str("sendWol", inputMacAddr).Msg("Invalid mac address provided")
```
~The `"sendWol"` is inconsistent with line 826 which is `"sendWOL"` (prefer that one)~

Actually, now that I've read the code again, it should be `"mac-addr"` since that's the name of the query-string parameter! (in both places)

IDisposable commented on web.go:
```diff
@@ -807,3 +810,23 @@ func handleSetup(c *gin.Context) {
+	macAddrString := macAddr.String()
+	err = rpcSendWOLMagicPacket(macAddrString)
+	if err != nil {
+		logger.Warn().Err(err).Str("sendWOL", macAddrString).Msg("Failed to send WOL magic packet")
```
Change `"sendWOL"` to `"mac-addr"`
